kotlin-audio/src/main/java/com/doublesymmetry/kotlinaudio/players/QueuedAudioPlayer.kt

```diff
@@ -74,12 +74,16 @@ class QueuedAudioPlayer(context: Context, bufferConfig: BufferConfig? = null, ca
      * @param playWhenReady If this is `true` it will automatically start playback. Default is `true`.
      */
     fun add(item: AudioItem, playWhenReady: Boolean = true) {
+        val hasCurrentItem = currentItem != null
         val mediaSource = getMediaSourceFromAudioItem(item)
         queue.add(mediaSource)
         exoPlayer.addMediaSource(mediaSource)
 
         exoPlayer.prepare()
-        exoPlayer.playWhenReady = playWhenReady
+
+        if (!hasCurrentItem && playWhenReady) {
```

Comment:
@mpivchev if there is currently an item playing, and `playWhenReady` is set/reset to `true` it causes the audio to stop playing (effectively a pause). By implementing a check it only sets it to true if there isn't anything already playing.

Perhaps there's a better fix but this is the one I found. The way this bug manifests is that if you have an empty queue then add an item to it, it will start playing as expected. If you then call `add` again, playback is paused. You can see the full steps to reproduce this on this ticket https://github.com/doublesymmetry/react-native-track-player/issues/1513

Reply:
So I'm currently testing this, and In the sample app it doesn't seem to stop playing if you call `add`. Did you only test this on RNTP?

Reply:
Yea, I tested in RNTP

Reply:
@mpivchev I just retested this in RNTP on the most recent commit and it is still pausing playback on `add`.

Reply:
I'm also able to reproduce this in the sample application by changing `onViewCreated` like this:
```diff
         player.add(firstItem)
-        player.add(secondItem)
         player.play()
+        player.add(secondItem, false)
```

What's the reason behind always setting `playWhenReady` when appending to the playlist by the way?
IMO it seems better to only exoPlayer.playWhenReady when `playWhenReady` is true and do nothing when it is false as it can interrupt the player.

Reply:
Just came across this issue after posting https://github.com/doublesymmetry/react-native-track-player/issues/1691 which shows that this occurs in a few places and is not only an Android issue. I think the solution is to remove all setting of `playWhenReady` (or pass whatever was previously set in `exoplayer.playWhenReady`).

Reply:
@puckey should I just close this PR and let you at it since you're addressing this on iOS as well?

Reply:
Sure thing!